lms/djangoapps/ccx/tests/test_overrides.py

```diff
@@ -129,3 +139,123 @@ def test_override_is_inherited_even_if_set_in_mooc(self):
         override_field_for_ccx(self.ccx, chapter, 'due', ccx_due)
         vertical = chapter.get_children()[0].get_children()[0]
         self.assertEqual(vertical.due, ccx_due)
+
+
+@attr('shard_1')
+@mock.patch.dict('django.conf.settings.FEATURES', {'ENABLE_XBLOCK_VIEW_ENDPOINT': True})
+@ddt.ddt
+class TestFieldOverridePerformance(ModuleStoreTestCase):
+    """
+    Tests that ensure the field
+    """
+    def setUp(self):
+        """
+        Create a test client, course, and user.
+        """
+        super(TestFieldOverridePerformance, self).setUp()
+
+        self.request_factory = RequestFactory()
+        self.student = UserFactory.create()
+        self.request = self.request_factory.get("foo")
+        self.request.user = self.student
+
+        MakoMiddleware().process_request(self.request)
+
+    def setup_course(self, course_name):
+        """
+        Imports some XML course data.
+        """
+        course = import_course_from_xml(
+            self.store,
+            999,
+            DATA_DIR,
+            ['test_increasing_size/graded_{}'.format(course_name)]
+        )[0]
+
+        CourseEnrollment.enroll(self.student, course.id)
+
+        return course
+
+    def grade_course(self, course):
+        """
+        Renders the progress page for the given course.
+        """
+        return progress(
+            self.request,
+            course_id=course.id.to_deprecated_string(),
+            student_id=self.student.id
+        )
+
+    def instrument_course_grading(self, course_name, queries, reads):
+        """
+        Renders the progress page, instrumenting Mongo reads and SQL queries.
+        """
+        course = self.setup_course(course_name)
+        with self.assertNumQueries(queries):
+            with check_mongo_calls(reads):
+                self.grade_course(course)
+
+    TEST_DATA = {
+        'xml': {
+            'no_overrides': [
+                (19, 7, 'small'), (23, 7, 'medium'), (27, 7, 'large')
+            ],
+            'ccx': [
+                (19, 24, 'small'), (23, 32, 'medium'), (27, 40, 'large')
+            ]
+        },
+        'split': {
+            'no_overrides': [
+                (19, 7, 'small'), (23, 7, 'medium'), (27, 7, 'large')
+            ],
+            'ccx': [
+                (20, 24, 'small'), (23, 32, 'medium'), (27, 40, 'large')
+            ]
+        }
+    }
+
+    @ddt.data(*TEST_DATA['xml']['no_overrides'])
```

Comment:
Rather than adding a new XML course to the test data, can you demonstrate the issue with a course constructed explicitly for these tests using `CourseFactory` and `ItemFactory`?